coral-hive/src/test/java/com/linkedin/coral/hive/hive2rel/HiveToRelConverterTest.java

```diff
@@ -622,6 +622,15 @@ public void testNameSakeColumnNamesShouldGetUniqueIdentifiers() {
     assertEquals(generated, expected);
   }
 
+  @Test
+  public void testUnionIntAndBigInt() {
+    final String sql = "SELECT a FROM test.tableOne UNION ALL SELECT z FROM test.tableTwo";
```

Comment:
(1) can we add a couple more assertions in the unit test - 
(int union all bigint), (bigint union all int), (tinyint union all bigint) etc

(2) could you add another UT (here & in HiveToTrino tests) for `CASE WHEN `table`.`a` = CAST(1 AS BIGINT)` type queries as mentioned in the PR description

Reply:
(1) The reason I originally didn't add more tests is because Calcite already [has tests](https://github.com/apache/calcite/blob/d5b6b5c01ff17cac100a591a4cc4c9e83d4e1ace/core/src/test/java/org/apache/calcite/sql/type/SqlTypeFactoryTest.java#L54) using the same precision values. But definitely agree more tests in Coral don't hurt. I added tests comparing INT to the other 3 integer types since those are the most common comparisons.

(2) Added.